web/backend/plugins/metrics.ts

```diff
@@ -1,10 +1,41 @@
 import { FastifyInstance } from 'fastify'
 import { getMetrics } from '../utils/metrics'
+import { join } from 'path'
+import { writeFile } from 'fs/promises'
+import { Mode } from '../schemas'
+import { getPidToLoad, getSelectableRuntimes } from '../utils/runtimes'
+import { RuntimeApiClient } from '@platformatic/control'
+import { MS_WAITING } from '../utils/constants'
 
 export default async function (fastify: FastifyInstance) {
+  const isValidMode = (mode: Mode) => fastify.loaded.mode === mode && prevMode !== mode
```

Comment:
I don't understand this helper. It seems there is a state machine at play, maybe model it as such?